src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntityChangeValidator.java

```diff
@@ -0,0 +1,86 @@
+package uk.gov.companieshouse.overseasentitiesapi.validation;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.model.company.RegisteredOfficeAddressApi;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.AddressDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.changes.*;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.commonmodels.Address;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.commonmodels.CompanyIdentification;
+
+@Component
+public class OverseasEntityChangeValidator {
```

Comment:
I'd view this more as a Comparator than a Validator and compare or checkChange rather than verify.